docs/reference/migration/migrate_2_0.asciidoc

```diff
@@ -138,6 +138,9 @@ In addition some query builders have been removed or renamed:
 * `filtered(...)` removed. Use `filteredQuery(...)` instead.
 * `inQuery(...)` removed.
 
+The methods IndicesQueryBuilder#parseNoMatchQuery(String) and IndicesQueryBuilder#noMatchQuery(String) have been removed,
```

Comment:
+1 on having MatchNoneQueryBuilder public, but without parser (like EmptyQueryBuilder). 
Also, I think having the String field in the builder (like `sNoMatchQuery` before) is not nice, we should only keep an inner QueryBuilder there. Maybe we could keep the old the `IndicesQueryBuilder#noMatchQuery(String)` method but restrict it to the two options previously offered in the parser (`all`, `none` I think), make this clear in javadocs, throw exceptions etc... and then immediately resolve the String input to `QueryBuilders#matchNoneQuery` or `QueryBuilders#matchAllQuery`? I don't mind as long as the builder doesn't keep the String argument around.